src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.FreeBSD.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace System.Diagnostics
+{
+    internal static partial class ProcessManager
+    {
+        /// <summary>Gets the IDs of all processes on the current machine.</summary>
+        public static int[] GetProcessIds()
+        {
+            return Interop.libutil.proc_listallpids();
+        }
+
+        /// <summary>Gets process infos for each process on the specified machine.</summary>
+        /// <param name="machineName">The target machine.</param>
+        /// <returns>An array of process infos, one per found process.</returns>
+        public static ProcessInfo[] GetProcessInfos(string machineName)
+        {
+            ThrowIfRemoteMachine(machineName);
+
+            int[] procIds = GetProcessIds(machineName);
+
+            // Iterate through all process IDs to load information about each process
+            var processes = new List<ProcessInfo>(procIds.Length);
+            foreach (int pid in procIds)
+            {
+                ProcessInfo pi = CreateProcessInfo(pid);
+                if (pi != null)
+                {
+                    processes.Add(pi);
+                }
+            }
+
+            return processes.ToArray();
+        }
+
+        // -----------------------------
+        // ---- PAL layer ends here ----
+        // -----------------------------
+
+        private static ProcessInfo CreateProcessInfo(int pid)
+        {
+            // Negative PIDs aren't valid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(pid));
+            }
+
+            ProcessInfo procInfo = new ProcessInfo()
+            {
+                ProcessId = pid
+            };
+
+            // Try to get the task info. This can fail if the user permissions don't permit
+            // this user context to query the specified process
+            ProcessInfo info = Interop.libutil.GetProcessInfoById(pid);
+            //if (info.HasValue)
+            if (true)
+            {
+                // Set the values we have; all the other values don't have meaning or don't exist
+                //unsafe { procInfo.ProcessName = Marshal.PtrToStringAnsi(new IntPtr(temp.pbsd.pbi_comm)); }
```

Comment:
Why is this commented out code here?

I'll stop commenting on such comments, but they apply throughout.